platforms/jvm/scala/src/main/java/org/gradle/api/plugins/scala/ScalaBasePlugin.java

```diff
@@ -96,27 +105,37 @@ public abstract class ScalaBasePlugin implements Plugin<Project> {
 
     private final ObjectFactory objectFactory;
     private final JvmPluginServices jvmPluginServices;
+    private final DependencyFactory dependencyFactory;
 
     @Inject
-    public ScalaBasePlugin(ObjectFactory objectFactory, JvmPluginServices jvmPluginServices) {
+    public ScalaBasePlugin(
+        ObjectFactory objectFactory,
+        JvmPluginServices jvmPluginServices,
+        DependencyFactory dependencyFactory
+    ) {
         this.objectFactory = objectFactory;
         this.jvmPluginServices = jvmPluginServices;
+        this.dependencyFactory = dependencyFactory;
     }
 
     @Override
     public void apply(final Project project) {
         project.getPluginManager().apply(JavaBasePlugin.class);
 
         ScalaRuntime scalaRuntime = project.getExtensions().create(SCALA_RUNTIME_EXTENSION_NAME, ScalaRuntime.class, project);
-        ScalaPluginExtension scalaPluginExtension = project.getExtensions().create(ScalaPluginExtension.class, "scala", DefaultScalaPluginExtension.class);
+
+        ScalaPluginExtension scalaPluginExtension = project.getExtensions().create(ScalaPluginExtension.class, "scala", ScalaPluginExtension.class);
```

Comment:
❗ This doesn't seem right - why is the `ScalaPluginExtension` an argument to the `ScalaPluginExtension`?

Reply:
This is not passing `ScalaPluginExtension.class` as an argument, it is using the `create` overload that accepts both a public type and an implementation type for an extension: https://github.com/gradle/gradle/blob/415c6c3cf3411348eb4ea673e4c6f2b5062911ea/subprojects/core-api/src/main/java/org/gradle/api/plugins/ExtensionContainer.java#L104C11-L104C20

However, I've simplified this to the overload that just takes an implementation type, since now the impl type == public type